app/gosqs/delete_message_batch.go

```diff
@@ -0,0 +1,114 @@
+package gosqs
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func DeleteMessageBatchV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+
+	requestBody := models.NewDeleteMessageBatchRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+
+	if !ok {
+		log.Error("Invalid Request - DeleteMessageBatchV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", true)
+	}
+
+	queueUrl := requestBody.QueueUrl
+
+	queueName := ""
+	if queueUrl == "" {
+		vars := mux.Vars(req)
+		queueName = vars["queueName"]
+	} else {
+		uriSegments := strings.Split(queueUrl, "/")
+		queueName = uriSegments[len(uriSegments)-1]
+	}
+
+	if _, ok := app.SyncQueues.Queues[queueName]; !ok {
+		return utils.CreateErrorResponseV1("QueueNotFound", true)
+	}
+
+	if len(requestBody.Entries) == 0 {
+		return utils.CreateErrorResponseV1("EmptyBatchRequest", true)
+	}
+
+	if len(requestBody.Entries) > 10 {
```

Comment:
Where did you find this requirement?  I don't see it in the docs or in the previous code, but I could be missing it.

Reply:
This is for "TooManyEntriesInBatchRequest" of the below:
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_DeleteMessageBatch.html

And the API abstraction says "Deletes up to ten messages from the specified queue." 👍

Reply:
Cool, I did not read that first line I guess haha.  Thank you for being thorough!